run/run.go

```diff
@@ -113,7 +228,9 @@ func mergeHeads(context *OctopusContext, remotes []git.LsRemoteEntry) ([]string,
 
 		context.Logger.Println("Trying simple merge with " + lsRemoteEntry.Ref)
 
-		_, err = context.Repo.Git("read-tree", "-u", "-m", "--aggressive", common, mrt, lsRemoteEntry.Sha1)
+		commonArray := strings.Split(common, "\n")
+		_, err = context.Repo.Git(append([]string{"read-tree", "-u", "-m", "--aggressive"},
+			append(commonArray, mrt, lsRemoteEntry.Sha1)...)...)
```

Comment:
Nice catch. Actually, it behaves the same way I was suggesting. The order of read-tree args is <mergeBase> <head> <remote>. The code you linked put head at max_args - 1. Then the 3 assignations are here https://github.com/git/git/blob/6e3a7b3398559305c7a239a42e447c21a8f39ff8/unpack-trees.c#L1720

* head at L1740. Takes args[max_args - 2] from your link.
* remote at L1720. Takes args[max_args - 1]
* mergeBase at L1739 surprisingly called index. Takes args[0]

We'd better put what I suggested rather than relying on this hidden behaviour.